### PR TITLE
Updated s3 directions, noted other cloud providers,  added warning

### PR DESCRIPTION
**Pull request by ekansa**
### brief description of changes
I updated the S3 example to reference the current version of the Python Boto3 library for interfacing with S3.

I also made a few edits to discuss other cloud storage providers so it doesn't read quite like an AWS advertisement. Please let me know if you think these changes properly reflect the current reality.

#### issues addressed
(https://github.com/archesproject/arches-docs/issues/323)

#### further comments
This change will need to applied to almost all branches.

---

This box **must** be checked
- [x] the PR branch was originally made from the base branch

This box **should** be checked
- [x] after these changes the docs build locally without error

This box **should only** be checked you intend to follow through on it (we can do it on our end too)
- [x] I will `cherry-pick` all commits in this PR into other branches that should have them _after_ this PR is merged

### CODE REVIEW

File: docs/using-aws-s3.txt [hunk 1]
```diff
@@ -125,7 +124,7 @@ app.  Before beginning, you will need to have set up and logged into your AWS ac
 
     .. code-block::
 
-      (ENV) $: pip install boto django-storages
+      (ENV) $: pip install boto3 django-storages
```
Comment (mradamcox) [inline docs/using-aws-s3.txt:127]: Putting explicit versions here, as per Arches Django release dependencies would be best in the long-run.

**Review by mradamcox — CHANGES_REQUESTED** [review]
Main thing to change is just the ### length. My other thoughts are more suggestions that probably shouldn't get in the way of merging this update.

File: docs/using-aws-s3.txt (hunk 1 above)
Comment (ekansa) [reply docs/using-aws-s3.txt:127]: If I'm reading this correctly (https://pypi.org/project/django-storages/) django storages  v 1.13.x is compatible with Django 3.2.x,

**Comment by ekansa** [comment]
@mradamcox Thanks for the feedback! I made the edits you requested in the PR

**Pull request merged**